### PR TITLE
integration with golang backend

### PR DESCRIPTION
**Pull request by ErykKul**
Aan de go-kant heb ik niet veel moeten doen (vooral dingen verwijderd): https://github.com/libis/rdm-integration/commit/dadc9e0fecd9f6beed4d10902cbdc1148cbc5ca0

Het gaat nog denk ik even duren voordat ik met Angular vlot kan werken. Toch een poging gedaan voor de integratie. Alle commentaar en/of verbeteringen zijn welkom.

### CODE REVIEW

**Comment by Kris-LIBIS** [comment]
The storage of the data in services is generally done to prevent APIs to be called whenever components et destroyed and recreated. However, it is not to be used for general application state. Storing the application state is what ngrx is created for and I would like to suggest that you have a look at that.

File: src/app/submit/submit.component.ts [hunk 1]
```diff
@@ -46,12 +41,21 @@ export class SubmitComponent implements OnInit {
...
-    this.data.data?.forEach(datafile => {
+    this.dataService.compare_result.data?.forEach(datafile => {
       if (datafile.action === Fileaction.Delete) {
         result.push(datafile);
       }
     })
     return result;
   }
 
+  submit() {
```
[the remark this answers is not in the document]
Comment (ErykKul) [reply src/app/submit/submit.component.ts:52]: Sounds good to me. I will implement that.

File: src/app/submit/submit.component.ts [hunk 2]
```diff
@@ -50,7 +50,17 @@ export class SubmitComponent implements OnInit {
...
+    this.dataService.compare_result.data?.forEach(datafile => {
+      let action = datafile.action === undefined ? Fileaction.Ignore : datafile.action;
+      if (action != Fileaction.Ignore) {
+        selected.push(datafile)
+      }
+    });
+    if (selected.length === 0) {
+      this.router.navigate(['/connect']);
+    }
+    this.dataService.submit(selected).subscribe((data: StoreResult) => {
```
[the remark this answers is not in the document]
Comment (ErykKul) [reply src/app/submit/submit.component.ts:64]: I have tried but failed so far... I will need some help with this one. As far as I can google, it is not necessary to unsubscribe from http client?

**Comment by ErykKul** [comment]
I like the status page idea and I will implement it. I will also look at the ngrx.
Furthermore, I did fix the getData thing with async pipe and the local variables, but it could still be a little weird, can you take a look at it again? I wanted to sort only once, and it looks more like the original code, so maybe it is fine like this?
I did not find an easy way to unsubscribe from the http client observable, maybe some code example could help? All the example I have found so far only do the "subscribe"...

**Comment by ErykKul** [comment]
I have found how to unsubscribe from the http client (I didn't know that the type of result of any subscribe call is a subscription object that you can call unsubscribe on...). I have also implemented the ngrx store for the credentials (that also needs subscribing, I implemented unsubscribe in the destructors). Next thing is the status page.

**Pull request merged**